src/writer/creator.cpp

```diff
@@ -164,12 +165,14 @@ namespace zim
 
     void Creator::addMetadata(const std::string& name, const std::string& content, const std::string& mimetype)
     {
+      checkError();
```

Comment:
It is not really justified (except maybe to avoid create the contentProvider if the creator is already in error state).
But I don't think it hurts to check error too much.